tests/e2e/common/parameterized_query_tests.py

```diff
@@ -0,0 +1,181 @@
+from datetime import datetime
+from decimal import Decimal
+
+import pytz
+
+from databricks.sql.client import Connection
+from databricks.sql.utils import DbSqlParameter, DbSqlType
+
+
+class PySQLParameterizedQueryTestSuiteMixin:
+    """Namespace for tests of server-side parameterized queries"""
+
+    def test_parameterized_query_named_and_inferred_e2e(self):
+        """Verify that named parameters passed to the database as a dict are returned of the correct type
+        All types are inferred.
+        """
+
+        conn: Connection
+
+        query = """
+        SELECT
+            :p_bool AS col_bool,
+            :p_int AS col_int,
+            :p_double AS col_double,
+            :p_date as col_date,
+            :p_timestamp as col_timestamp,
+            :p_str AS col_str
+        """
+
+        named_parameters = {
+            "p_bool": True,
+            "p_int": 1234,
+            "p_double": 3.14,
+            "p_date": datetime(2023, 9, 6, 3, 14, 27, 843, tzinfo=pytz.UTC).date(),
+            "p_timestamp": datetime(2023, 9, 6, 3, 14, 27, 843, tzinfo=pytz.UTC),
+            "p_str": "Hello",
+        }
+        with self.connection() as conn:
```

Comment:
I could be mistaken, but this block looks duplicated below.  Good idea to DRY it up, so if you need to make changes in future, you can make them in one place.